src/Plug.vala

```diff
@@ -378,20 +378,22 @@ namespace Power {
         private static bool lid_detect () {
             var interface_path = File.new_for_path ("/proc/acpi/button/lid/");
 
-            try {
-                var enumerator = interface_path.enumerate_children (
-                GLib.FileAttribute.STANDARD_NAME,
-                FileQueryInfoFlags.NONE);
-                FileInfo lid;
-                if ((lid = enumerator.next_file ()) != null) {
-                    debug ("Detected lid switch");
-                    return true;
-                }
+            if (interface_path.query_exists()) {
```

Comment:
Space before brackets: `query_exists ()`.

Reply:
I wonder about that check in the backlight_detect.

The /sys/class/backlight is kernel sysfs interface which should be always present.
It only has entries when backlight driver is loaded.